opynfinance/squeeth-monorepo
# Otc Hedge

- Init crab v2
- Port v1 tests to v2
- Change CrabStrategyV2 typechain
- Added otchedge code without test; and removed uniswap/price hedge
- tests duplicated for otc hedge; needs cleanup
- renamed managerBuyPrice variable name
- reduce trader sell quantity when buy price is greater
- added time and price threshold checks
- Fix: ordered the var updates
- added increasing price check & removed auction uniswap code
- Removed auctions
- made the tests pending for the code removed
- removed auctionTriggerTime from isTimeHege
- price check for OTC price vs uniswap
- ran prettier on crab test hege otc
- ran prettier on crab contract hege otc
- lint
- Check all trade pairs are the same
- added multiple orders as test
- added test for single order, multiple orders from different people
- added buyside test for otchedge
- added checks for traders balance as well
- nit: trader is now a variable
- organized the multiple orders test
- made one testcase more readable with functions
- added test for manager being able to trade in lesser quantity
- added test for manager to pass a large sellAmount
- added test to allow the manager to buy at a greater price
- added test for lesser quantity and greater price
- Added test for timeThreshold and priceThresholds
- ran formatter on test
- tests for check otc price
- refactored type and domain data
- added test for invalid signature i.e a fake order
- test for manager buy price atleast seller price
- tests for manager buyer price should be greater than 0
- added test for correct formatting of orders
- added tests for setters and re arranged the crab contract
- Add vault migration with timelock logic (#530)
- 
- * added expiries
- 
- * updated test orders with t+ 10 min expiries
- 
- * fixed typehash
- 
- * added expiry to type data in test file
- 
- * orders with prices
- 
- * fixed direction of remaining quantity che k
- 
- * fix natspec
- 
- * reduced remainingAmounts in hedgeOTC
- 
- * checks that orders are all in same direction and trading opposite to hedge
- 
- * passing in flags for _mintWPowerPerp and _burnPowerPerp + naming + revert messages
- 
- * isHedgeBuying vs isOrderBuying
- 
- * fixes and lint
- 
- * added withdrawing weth, initialize current price; fixed 3 tests;
- 
- * fixed test for allowing to trade lower quantity
- 
- * formatted the document
- 
- * fixed test for allowing vault manager to trade more quantity
- 
- * fixed test for allowing vault manager to buy at greater price
- 
- * fixed test for allowing vault manager to buy at greater price and specify lesser quanitity
- 
- * fixed test for high price by manager
- 
- * fixed test for invalid signature
- 
- * fixed test for manager price < order price
- 
- * fixed test for ordering of orders
- 
- * removed import console
- 
-
